File: backend/src/main/java/com/tmk/vtcmanager/application/domain/penalite/LignePenaliteFiltres.java
package com.tmk.vtcmanager.application.domain.penalite;

import com.tmk.vtcmanager.application.domain.conditionTravail.TypeSanction;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;

@Data
@Builder
public class LignePenaliteFiltres {
    private Long vehiculeId;
    private Long chauffeurId;
    private TypeSanction typeSanction;
    private StatutLignePenalite statut;
    private LocalDate dateDebut;
    private LocalDate dateFin;
}

File: mobile/lib/features/penalite/domain/entities/ligne_penalite_filtres.dart
import 'ligne_penalite.dart';

class LignePenaliteFiltres {
  final int? vehiculeId;
  final int? chauffeurId;
  final TypeSanctionLigne? typeSanction;
  final StatutLignePenalite? statut;
  final DateTime? dateDebut;
  final DateTime? dateFin;

  /// Mot-clé libre : immatriculation du véhicule ou nom/prénom du chauffeur.
  /// Évalué côté serveur, sur l'ensemble des lignes et non la page chargée.
  final String? recherche;

  const LignePenaliteFiltres({
    this.vehiculeId,
    this.chauffeurId,
    this.typeSanction,
    this.statut,
    this.dateDebut,
    this.dateFin,
    this.recherche,
  });
}
